client/containers/Pub/PubSyncManager.tsx

```diff
@@ -112,6 +116,8 @@ const idleStateUpdater = (boundSetState, timeout = 50) => {
 
 type State = any;
 class PubSyncManager extends React.Component<Props, State> {
+	idleStateUpdater: ReturnType<typeof idleStateUpdater>;
```

Comment:
I see you've used `typeof x` in a couple places here and it made me wonder if you have any thoughts on explicitly typing with a type alias or interface vs deriving types with `typeof`. I've tried to avoid using `typeof` to derive types in the past (only using it when necessary, like if a library doesn't export an alias or interface), but I wonder if it's really worth the effort to provision an interface explicitly it if the subject is unlikely to be used polymorphically (is that a word?).

Reply:
I really like using `ReturnType<typeof fn>`. The other usage in this file I think is more out of laziness. But I have a sense that we're going to pick apart and completely re-implement this file, which is getting way too big for its britches, in the next six months or so...so I'm being a little loose about these things for now.